src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -883,6 +883,72 @@ class spell_dru_ferocious_bite : public SpellScript
     float _damageMultiplier = 0.0f;
 };
 
+// 197721 - Flourish
+class spell_dru_flourish : public SpellScript
+{
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        targets.remove_if([&](WorldObject const* target)
+        {
+            return !IsTargetValidForFlourish(target);
+        });
+    }
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/) const
+    {
+        auto const it = _targetAndAuras.find(GetHitUnit()->GetGUID());
+        if (it == _targetAndAuras.end())
+            return;
+
+        Milliseconds const extraDuration = Seconds(GetEffectValue());
+
+        for (Aura* aura : it->second)
+        {
+            aura->SetDuration(aura->GetDuration() + extraDuration.count());
+            aura->SetMaxDuration(aura->GetMaxDuration() + extraDuration.count());
+        }
+    }
+
+    void Register() override
+    {
+        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_dru_flourish::FilterTargets, EFFECT_0, TARGET_UNIT_DEST_AREA_ALLY);
+        OnEffectHitTarget += SpellEffectFn(spell_dru_flourish::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+
+private:
+    bool IsTargetValidForFlourish(WorldObject const* target)
+    {
+        Unit const* unitTarget = target->ToUnit();
+        if (!unitTarget)
+            return false;
+
+        bool hasValidAura = false;
+
+        for (AuraEffect const* auraEff : unitTarget->GetAuraEffectsByType(SPELL_AURA_PERIODIC_HEAL))
+        {
+            Aura* aura = auraEff->GetBase();
+            if (!aura || aura->GetCasterGUID() != GetCaster()->GetGUID())
+                continue;
+
+            SpellInfo const* auraInfo = aura->GetSpellInfo();
+            if (!auraInfo || auraInfo->SpellFamilyName != SPELLFAMILY_DRUID)
+                continue;
+
+            RegisterAffectedAurasForTarget(unitTarget->GetGUID(), aura);
+            hasValidAura = true;
```

Comment:
not sure if it'd be any better in terms of performance to check for the value to be false to be set to true, should be the same overhead but whatever